### ISSUE TITLE
How to calculate the average euclidean distance in the paper?

### ISSUE THREAD

**Issue by kenmbkr** [AliaksandrSiarohin/first-order-model#148]
The metrics part of the experiment section says

> We employ the feature embedding used in Monkey-Net.

I looked up the Monkey-Net paper and it says

> We compute the feature-based metric employed in [13] that consists in computing the Average Euclidean Distance (AED) ... More  precisely,  we  use  a  network trained for facial identification [1] for Nemo and a network trained for person re-id [19] for Tai-Chi.

Reference 13 in the Monkey-Net is the [Variational U-Net](https://arxiv.org/abs/1804.04694) paper, but it uses `std` and `max pairwise dist` to measure the features. Those two metrics have different scales compared to the `AED` in the Monkey-Net and first-order papers.

Based on the description in the Monkey-Net paper, there seems be one extra network for each dataset for feature extraction. It is still not clear how to obtain the `AED` in both papers.

Could you elaborate on the methodology to calculate the `AED`?

**Comment by AliaksandrSiarohin** [comment]
You can use the code from here https://github.com/AliaksandrSiarohin/pose-evaluation

**Comment by zhhezhhe** [comment]
I tried the code in the link you pointed to, I got 401.656346094842 on AED metric on VoxCeleb dataset, in the paper, the reported result is 0.140, how can I match my result with the reported result?

**Comment by AliaksandrSiarohin** [comment]
What commands you are running and what version of pytorch are you using?

**Comment by AliaksandrSiarohin** [comment]
What commands you are running and what version of pytorch are you using?

**Comment by zhhezhhe** [comment]
```
pytorch1.0.0
python3.6

python extract.py --in_folder gt/png --out_file id_gt.pkl --is_video --type face_pose --image_shape 256,256
python extract.py --in_folder reconstruction/png --out_file result.pkl --is_video --type face_pose --image_shape 256,256
python cmp.py id_gt.pkl result.pkl
```

**Comment by AliaksandrSiarohin** [comment]
face_pose is for AKD, you shoud use face_id. I update the README.md

**Comment by zhhezhhe** [comment]
Thank you for the reply.
